### ISSUE TITLE
CHF credit transfer to QR-IBAN

### ISSUE THREAD

**Issue by andre-m-dev** [viafintech/sps_king#9]
We are experiencing problems with credit transfer to QR-IBANs. It seems that the code/scheme has no understanding of QR-IBANs.

The pain xml should look like this (relevant excerpt):
```
<?xml version="1.0" encoding="UTF-8"?>
<Document xmlns="urn:iso:std:iso:20022:tech:xsd:pain.001.001.09" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="urn:iso:std:iso:20022:tech:xsd:pain.001.001.09  pain.001.001.09.ch.03.xsd">
	<CstmrCdtTrfInitn>
		<PmtInf>
			<CdtTrfTxInf>
				<PmtId>
					<InstrId>INSTRID-01-01</InstrId>
					<EndToEndId>ENDTOENDID-QRR</EndToEndId>
				</PmtId>
				<Amt>
					<InstdAmt Ccy="CHF">3949.75</InstdAmt>
				</Amt>
				<CdtrAcct>
					<Id>
						<IBAN>CH4431999123000889012</IBAN>
					</Id>
				</CdtrAcct>
				<RmtInf>
					<Strd>
						<CdtrRefInf>
							<Tp>
								<CdOrPrtry>
									<Prtry>QRR</Prtry>
								</CdOrPrtry>
							</Tp>
							<Ref>210000000003139471430009017</Ref>
						</CdtrRefInf>
						<AddtlRmtInf>Auftrag vom 10.02.2023</AddtlRmtInf>
					</Strd>
				</RmtInf>
			</CdtTrfTxInf>
		</PmtInf>
	</CstmrCdtTrfInitn>
</Document>
```

The missing piece of the puzzle will probably be `<Prtry>QRR</Prtry>`

If an implementation is necessary, I can definitely test the change.

**Comment by tobischo** [comment]
Could you elaborate on the difference between an IBAN and a QR-IBAN?

Shouldn't the QR code just be the encoded version of the information, e.g. the IBAN, the remittance text?

**Comment by tobischo** [comment]
I guess it is important for the structured remittance information to be understood as being a QR code?

That part should then be relatively easy. The questions is whether you also need pain.001.001.09 or if you can still live with pain.001.001.03, which is currently implemented

**Comment by andre-m-dev** [comment]
> I guess it is important for the structured remittance information to be understood as being a QR code?

Yes
 
> That part should then be relatively easy. The questions is whether you also need pain.001.001.09 or if you can still live with pain.001.001.03, which is currently implemented

If pain.001.001.03 is compatible with the changes, then it can stay as it is

**Comment by tobischo** [comment]
I prepared the change with only QRR in #10 Feel free to test it. Let me know if you *need* pain.001.001.09 as part of this.

Otherwise I would rather leave that for a later change

**Comment by tobischo** [comment]
NVM, I think I forgot something 😄 

So far the structured remittance information is only supported for debits (we never needed it for credits)

**Comment by andre-m-dev** [comment]
That's why I was also confused by the documentation in the Readme, because structured_remittance_information is only listed for direct debit

**Comment by tobischo** [comment]
Alright, so this allows you to generate files like this:

```xml
<?xml version="1.0"?>
<Document xmlns="http://www.six-interbank-clearing.com/de/pain.001.001.03.ch.02.xsd" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.six-interbank-clearing.com/de/pain.001.001.03.ch.02.xsd pain.001.001.03.ch.02.xsd">
  <CstmrCdtTrfInitn>
    <GrpHdr>
      <MsgId>SPS-KING/218d930bc0f46dd5c00645</MsgId>
      <CreDtTm>2024-07-31T16:58:46+02:00</CreDtTm>
      <NbOfTxs>1</NbOfTxs>
      <CtrlSum>102.50</CtrlSum>
      <InitgPty>
        <Nm>Schuldner GmbH</Nm>
      </InitgPty>
    </GrpHdr>
    <PmtInf>
      <PmtInfId>SPS-KING/218d930bc0f46dd5c00645/1</PmtInfId>
      <PmtMtd>TRF</PmtMtd>
      <BtchBookg>true</BtchBookg>
      <NbOfTxs>1</NbOfTxs>
      <CtrlSum>102.50</CtrlSum>
      <PmtTpInf/>
      <ReqdExctnDt>1999-01-01</ReqdExctnDt>
      <Dbtr>
        <Nm>Schuldner GmbH</Nm>
      </Dbtr>
      <DbtrAcct>
        <Id>
          <IBAN>CH5481230000001998736</IBAN>
        </Id>
      </DbtrAcct>
      <DbtrAgt>
        <FinInstnId>
          <BIC>RAIFCH22</BIC>
        </FinInstnId>
      </DbtrAgt>
      <CdtTrfTxInf>
        <PmtId>
          <EndToEndId>NOTPROVIDED</EndToEndId>
        </PmtId>
        <Amt>
          <InstdAmt Ccy="CHF">102.50</InstdAmt>
        </Amt>
        <CdtrAgt>
          <FinInstnId>
            <BIC>RAIFCH22</BIC>
          </FinInstnId>
        </CdtrAgt>
        <Cdtr>
          <Nm>Contoso AG</Nm>
        </Cdtr>
        <CdtrAcct>
          <Id>
            <IBAN>CH5481230000001998736</IBAN>
          </Id>
        </CdtrAcct>
        <RmtInf>
          <Strd>
            <CdtrRefInf>
              <Tp>
                <CdOrPrtry>
                  <Prtry>QRR</Prtry>
                </CdOrPrtry>
              </Tp>
              <Ref>185744810000000000200800628</Ref>
            </CdtrRefInf>
          </Strd>
        </RmtInf>
      </CdtTrfTxInf>
    </PmtInf>
  </CstmrCdtTrfInitn>
</Document>
```

**Comment by andre-m-dev** [comment]
@tobischo we were able to successfully test the change with our bank in Switzerland. Many thanks again.

**Comment by tobischo** [comment]
Released as v0.5.0
